src/alchemlyb/parsing/namd.py

```diff
@@ -25,72 +82,136 @@ def extract_u_nk(fep_file, T):
+    # The assumption is that they make sense in lexicographic order.
+    for fep_file in sorted(fep_files):
+        lambda1_at_start, lambda2_at_start, lambda_idws_at_start = None, None, None
+        # Note we have not set parsing=False because we could be continuing one window across
+        # more than one fepout file
+        with anyopen(fep_file, 'r') as f:
+            has_idws = False
+            for line in f:
+                l = line.strip().split()
+                # We don't know if IDWS was enabled just from the #Free line, and we might not have
+                # a #NEW line in this file, so we have to check for the existence of FepE_back lines
+                # We rely on short-circuit evaluation to avoid the string comparison most of the time
+                if has_idws is False and l[0] == 'FepE_back:':
+                    has_idws = True
+
+                # New window, get IDWS lambda if any
+                # We keep track of lambdas from the #NEW line and if they disagree with the #Free line
+                # within the same file, then complain. This can happen if truncated fepout files
+                # are presented in the wrong order.
+                if l[0] == '#NEW':
+                    lambda1_at_start, lambda2_at_start = float(l[6]), float(l[8])
+                    lambda_idws_at_start = float(l[10]) if 'LAMBDA_IDWS' in l else None
+                    has_idws = True if lambda_idws_at_start is not None else False
+
+                # this line marks end of window; dump data into dataframe
+                if l[0] == '#Free':
+                    # extract lambda values for finished window
+                    # lambda1 = sampling lambda (row), lambda2 = comparison lambda (col)
+                    lambda1 = float(l[7])
+                    lambda2 = float(l[8])
+                    lambda1_idx = all_lambdas.index(lambda1)
+                    if has_idws is True and lambda1_idx > 0:
+                        lambda_idws = all_lambdas[lambda1_idx - 1]
+                    else:
+                        lambda_idws = None
+
+                    # If the lambdas are not what we thought they would be, return None, ensuring the calculation
+                    # fails. This can happen if fepouts where one window spans multiple fepouts are processed out of order
+                    if lambda1_at_start is not None \
+                        and (lambda1, lambda2, lambda_idws) != (lambda1_at_start, lambda2_at_start, lambda_idws_at_start):
+                        print("namd.py: extract_u_nk: Error: Lambdas changed unexpectedly while processing", fep_file)
+                        print(f"namd.py: extract_u_nk: Error: l1, l2, lidws: {lambda1_at_start}, {lambda2_at_start}, {lambda_idws_at_start} changed to {lambda1}, {lambda2}, {lambda_idws}")
+                        print(f"namd.py: extract_u_nk: Error: fep_file = {fep_file}; has_idws = {has_idws}")
+                        return None
```

Comment:
Needs to be tested, check the coverage report (maybe the link https://app.codecov.io/gh/alchemistry/alchemlyb/compare/135/diff#diff-c3JjL2FsY2hlbWx5Yi9wYXJzaW5nL25hbWQucHk= works)

Reply:
@orbeckst That link gives me an "GitHub API rate limit error" message. I'm unfamiliar with Codecov, but I can at least add a test for this

Reply:
Hm, yes, I get that, too, when I am not logged in with my GitHub account.

Look at the running tests and then click on the codecov actions (e.g., https://github.com/alchemistry/alchemlyb/pull/135/checks?check_run_id=3314785981 ) and then go from there?

Reply:
This leads me to https://app.codecov.io/gh/alchemistry/alchemlyb/compare/135/diff — when I tried this link in an anonynmous browser window without being logged in anywhere I could see everything.

Reply:
Thanks. I am not familiar with Codecov so I'll need to figure out how to tell it what my still-to-be-written new tests cover.